File: tmp_fix_routing.py
import re
import sys

def main():
    file_path = r"e:\MyProject\GIT\caj-store-app\src\components\Header.tsx"
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Check if useRouter is imported
    if "useRouter" not in content:
        # Import useRouter
        content = content.replace("import Link from 'next/link';", "import Link from 'next/link';\nimport { useRouter } from 'next/navigation';")
        
    # Inject router initialization inside Header component
    if "const router = useRouter();" not in content:
        content = content.replace("export default function Header() {", "export default function Header() {\n  const router = useRouter();")

    # Replace <DropdownMenuItem asChild><Link href="...">...</Link></DropdownMenuItem>
    # Note: Regex to handle arbitrary nested things inside Link.
    pattern = r'<DropdownMenuItem asChild><Link href="([^"]+)"([^>]*)>(.*?)<\/Link><\/DropdownMenuItem>'
    
    def replacer(match):
        href = match.group(1)
        # className and other props on Link
        props = match.group(2)
        inner = match.group(3)
        return f'<DropdownMenuItem onSelect={{() => router.push("{href}")}}{props}>{inner}</DropdownMenuItem>'

    new_content = re.sub(pattern, replacer, content, flags=re.DOTALL)
    
    # Now for <DropdownMenuItem asChild><Link href="/admin/subscription" ...>
    # There are also multiline ones like:
    # <DropdownMenuItem asChild>
    #   <Link href="/admin/subscription" className="flex items-center gap-2 cursor-pointer">
    #     <CreditCard className="h-4 w-4" /> Subscription
    #   </Link>
    # </DropdownMenuItem>
    pattern_multi = r'<DropdownMenuItem asChild>\s*<Link href="([^"]+)"([^>]*)>\s*(.*?)\s*<\/Link>\s*<\/DropdownMenuItem>'
    new_content = re.sub(pattern_multi, replacer, new_content, flags=re.DOTALL)

    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(new_content)
        
    print("Routing references updated!")

if __name__ == "__main__":
    main()

File: tmp_fix_routing.js
const fs = require('fs');
const path = require('path');

function main() {
    const filePath = path.join(__dirname, 'src', 'components', 'Header.tsx');
    let content = fs.readFileSync(filePath, 'utf-8');

    if (!content.includes('useRouter')) {
        content = content.replace("import Link from 'next/link';", "import Link from 'next/link';\nimport { useRouter } from 'next/navigation';");
    }

    if (!content.includes('const router = useRouter();')) {
        content = content.replace("export default function Header() {", "export default function Header() {\n  const router = useRouter();");
    }

    // Replace single line matches
    let newContent = content.replace(/<DropdownMenuItem asChild><Link href="([^"]+)"([^>]*)>(.*?)<\/Link><\/DropdownMenuItem>/gs, (match, href, props, inner) => {
        return `<DropdownMenuItem onSelect={() => router.push("${href}")}${props}>${inner}</DropdownMenuItem>`;
    });

    // Replace multiline matches
    newContent = newContent.replace(/<DropdownMenuItem asChild>\s*<Link href="([^"]+)"([^>]*)>\s*(.*?)\s*<\/Link>\s*<\/DropdownMenuItem>/gs, (match, href, props, inner) => {
        return `<DropdownMenuItem onSelect={() => router.push("${href}")}${props}>\n${inner}\n</DropdownMenuItem>`;
    });

    fs.writeFileSync(filePath, newContent, 'utf-8');
    console.log("Routing references updated via Node!");
}

main();
